src/mlpack/methods/ann/layer/isru.hpp

```diff
@@ -0,0 +1,123 @@
+ *
+ * For more information, read the following paper (page 6).
+ *
+ * @code
+ * @article{
+ *   author  = {Carlile, Brad and Delamarter, Guy and Kinney, Paul and Marti,
+ *              Akiko and Whitney, Brian},
+ *   title   = {Improving deep learning by inverse square root linear units (ISRLUs)},
+ *   year    = {2017},
+ *   url     = {https://arxiv.org/pdf/1710.09967.pdf}
+ * }
+ * @endcode
+ * 
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_ISRU_HPP
+#define MLPACK_METHODS_ANN_LAYER_ISRU_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The ISRU activation function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) &=& x(\frac{1}{1 + \alpha x^2}) \\
+ * f'(x) &=& \frac{1}{1 + \alpha x^2}^3 \\
+ * @f}
+ *
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template<typename InputDataType = arma::mat,
+         typename OutputDataType = arma::mat>
+class ISRU
+{
+ public:
+
+ /**
+   * Create the ISRU object using the specified parameter.
+   *
+   * @param alpha Scale parameter.
+  */
+  ISRU(const double alpha = 1.0);
+
+  /**
+   * Ordinary feed forward pass of a neural network, evaluating the function
+   * f(x) by propagating the activity forward through f.
+   *
+   * @param input Input data used for evaluating the specified function.
+   * @param output Resulting output activation.
+   */
+  template<typename InputType, typename OutputType>
+  void Forward(const InputType& input, OutputType& output);
+
+  /**
+   * Ordinary feed backward pass of a neural network, calculating the function
+   * f(x) by propagating x backwards through f. Using the results from the feed
+   * forward pass.
+   *
+   * @param input The propagated input activation.
+   * @param gy The backpropagated error.
+   * @param g The calculated gradient.
+   */
+  template<typename DataType>
+  void Backward(const DataType& input, const DataType& gy, DataType& g);
+
+  //! Get the output parameter.
+  OutputDataType const& OutputParameter() const { return outputParameter; }
+  //! Modify the output parameter.
+  OutputDataType& OutputParameter() { return outputParameter; }
+
+  //! Get the delta.
+  OutputDataType const& Delta() const { return delta; }
+  //! Modify the delta.
+  OutputDataType& Delta() { return delta; }
+  
+    //! Get the non zero gradient.
+  double const& Alpha() const { return alpha; }
+  
+  //! Modify the non zero gradient.
+  double& Alpha() { return alpha; }
```

Comment:
In the constructor you mentioned it's for scaling, I think both is correct but maybe we should be a little bit more consistent?

Reply:
Oh ok...so should I replace the descriptions with "positive scale parameter" instead for consistency?

Reply:
Sounds good to me.